### DISCUSSION TITLE
Errors in the release notes for v4.2.0-beta2

### DISCUSSION THREAD

**Discussion by rgsteele** [mastodon/mastodon#26775] [General]
I noted what I believe are errors in the release notes for v4.2.0-beta2. I hope this is the right place to report these.

First, under **Streaming server changes**, this command:

    sudo cp ~mastodon/live/dist/mastodon-streaming*.service /etc/systemd/system/

This won't work as wildcard expansion happens before `sudo` is executed. Perhaps this instead:

    for i in "" "@"; do sudo cp ~mastodon/live/dist/mastodon-streaming${i}.service /etc/systemd/system/; done

Second, under **Update steps > Non-docker only**, it says:

>If you are using rbenv, [update the list of available versions](https://github.com/rbenv/rbenv#updating-the-list-of-available-ruby-versions) and install Ruby 3.2.2 by doing `RUBY_CONFIGURE_OPTS=--with-jemalloc rbenv install`

The version number is missing from the command. It should be: `RUBY_CONFIGURE_OPTS=--with-jemalloc rbenv install 3.2.2`.

Also, the [installation instructions](https://docs.joinmastodon.org/admin/install/#installing-ruby) specify to set the global Ruby version with `rbenv global 3.0.6`. I would assume this should be done again for the new version?

Finally, it would be helpful to have more detailed instructions for "Restart all Mastodon" processes. I ran `sudo systemctl restart "mastodon-*"` and then tried running the post-deployment migration step, but that failed, I guess because not everything was up and running yet. I tried it again a few minutes later and it seemed to succeed, but my instance was still not accessible. I rebooted the server and went to bed and it was up and running when I woke up the next morning, but some more explicit instructions about how to restart the processes and what logs to watch to know when startup is complete would be useful for impatient noobs like me 😃

**Comment by renchap** [comment]
@ClearlyClaire this one is for you!

**Comment by ClearlyClaire** [comment]
> First, under **Streaming server changes**, this command:
> 
> ```
> sudo cp ~mastodon/live/dist/mastodon-streaming*.service /etc/systemd/system/
> ```
> 
> This won't work as wildcard expansion happens before `sudo` is executed.

But the wildcard expansion happening before `sudo` is executed should not matter, as it should expand to the same thing regardless of whether you are running this as `mastodon` or `root`, right?

> > If you are using rbenv, [update the list of available versions](https://github.com/rbenv/rbenv#updating-the-list-of-available-ruby-versions) and install Ruby 3.2.2 by doing `RUBY_CONFIGURE_OPTS=--with-jemalloc rbenv install`
> 
> The version number is missing from the command. It should be: `RUBY_CONFIGURE_OPTS=--with-jemalloc rbenv install 3.2.2`.

If run in `/home/mastodon/live` and the version string is omitted from the command-line, it will use `.ruby-version`, which is what we want. Maybe it would make sense to have the release notes be more explicit about running it from the Mastodon installation directory.

> Also, the [installation instructions](https://docs.joinmastodon.org/admin/install/#installing-ruby) specify to set the global Ruby version with `rbenv global 3.0.6`. I would assume this should be done again for the new version?

Similarly to the previous comment, running from `/home/mastodon/live` will pick up the correct version. I think the install instructions should avoid mentioning the version, to be more future-proof.

> and then tried running the post-deployment migration step, but that failed, I guess because not everything was up and running yet

Do you still have logs from there? Running the post-deployment migration steps should not require any Mastodon service to be running.

**Comment by rgsteele** [comment]
> > First, under **Streaming server changes**, this command:
> > ```
> > sudo cp ~mastodon/live/dist/mastodon-streaming*.service /etc/systemd/system/
> > ```
> > This won't work as wildcard expansion happens before `sudo` is executed.
> 
> But the wildcard expansion happening before `sudo` is executed should not matter, as it should expand to the same thing regardless of whether you are running this as `mastodon` or `root`, right? (Running as `mastodon` won't work as it is not in 

The presence of `sudo` in the command led me to assume that it was to be run as a standard user, who would not have access to `~mastodon`, preventing the expansion. Another option of course would be to run as root; but in that case, `sudo` wouldn't be required as part of the command. (Running as `mastodon` won't work unless that account has been added to sudoers or a password has been set on `root`.)

> If run in `/home/mastodon/live` and the version string is omitted from the command-line, it will use `.ruby-version`, which is what we want. Maybe it would make sense to have the release notes be more explicit about running it from the Mastodon installation directory.

Yes, I must have been in the wrong directory. Thanks for the clarification.

> > and then tried running the post-deployment migration step, but that failed, I guess because not everything was up and running yet
> 
> Do you still have logs from there? Running the post-deployment migration steps should not require any Mastodon service to be running.

I don't, sorry. I expect I must have mis-attributed the failure and something else was going on.
